### PR TITLE
DDF-1489 Fixed OpenSearchSource and added ITest

### PR DESCRIPTION
**Pull request by harrison-tarr**
Added logic to accept purely spatial queries to OpenSearchSource
and added a wildcard character as a contextual filter since
OpenSearch needs it. In addition, added an ITest to ensure pure
spatial federated OpenSearches work.

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/175)

### CODE REVIEW

**Comment by harrison-tarr** [comment]
@shaundmorris @brendan-hofmann @figliold @pklinef

**Comment by brendan-hofmann** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 429 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/175#-JzGTZnqTG60Z-qBJbFt)** ([raw file](https://github.com/codice/ddf/blob/c356b4c865a7dfeb778ed5a46ce624cf147f5887/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L429)):</sup>
This block should use parameterized logging without the conditional wrapper. Removing the toString calls (since slf4j already does that for you) eliminates any performance impact.

See http://www.slf4j.org/faq.html#logging_performance and http://www.slf4j.org/faq.html#string_contents

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 444 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/175#-JzGTnJC2ZHOxyze8uFv)** ([raw file](https://github.com/codice/ddf/blob/c356b4c865a7dfeb778ed5a46ce624cf147f5887/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L444)):</sup>
These two exceptions probably shouldn't be logged at info. Warn if they're important, Debug if they're expected.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/175)

**Comment by harrison-tarr** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 429 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/175#-JzGTZnqTG60Z-qBJbFt)** ([raw file](https://github.com/codice/ddf/blob/c356b4c865a7dfeb778ed5a46ce624cf147f5887/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L429)):</sup>
in response to both of your comments: this was code i copied and pasted. should I make these changes in both the copied code and the pasted code, just mine, or neither (to keep it consistent)? I imagine both, but i just want to double check

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/175)

File: catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java [hunk 1]
```diff
@@ -413,6 +413,53 @@ protected boolean setOpenSearchParameters(Query query, Subject subject, WebClien
...
+
+            OpenSearchSiteUtil.populateSearchOptions(client, query, subject, parameters);
+
+            // add a wildcard search term - this query came in with no search phrase and a search phrase is necessary
+            OpenSearchSiteUtil.populateContextual(client, "*", parameters);
+
+            TemporalFilter temporalFilter = visitor.getTemporalSearch();
+            if (temporalFilter != null) {
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("startDate = " + temporalFilter.getStartDate().toString());
```
Comment (AzGoalie) [inline catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java]: Should use parameterized logging for all your logging.

File: catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java [hunk 2]
```diff
@@ -413,6 +413,53 @@ protected boolean setOpenSearchParameters(Query query, Subject subject, WebClien
...
+                contextualFilter == null)) {
+
+            OpenSearchSiteUtil.populateSearchOptions(client, query, subject, parameters);
+
+            // add a wildcard search term - this query came in with no search phrase and a search phrase is necessary
+            OpenSearchSiteUtil.populateContextual(client, "*", parameters);
+
+            TemporalFilter temporalFilter = visitor.getTemporalSearch();
+            if (temporalFilter != null) {
+                if (LOGGER.isDebugEnabled()) {
```
Comment (AzGoalie) [inline catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java]: If you use parameterized logging, you don't need this.

**Comment by brendan-hofmann** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 429 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/175#-JzGTZnqTG60Z-qBJbFt)** ([raw file](https://github.com/codice/ddf/blob/c356b4c865a7dfeb778ed5a46ce624cf147f5887/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L429)):</sup>
It's a very low-risk change, so I'd say fix both. ;)

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/175)

**Comment by brendan-hofmann** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 429 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/175#-JzGTZnqTG60Z-qBJbFt)** ([raw file](https://github.com/codice/ddf/blob/c356b4c865a7dfeb778ed5a46ce624cf147f5887/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L429)):</sup>
It would be nice if we could refactor instead of duplicating code, but that might be out of scope, depending on how difficult it would be.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/175)

**Comment by brendan-hofmann** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 429 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/175#-JzGTZnqTG60Z-qBJbFt)** ([raw file](https://github.com/codice/ddf/blob/c356b4c865a7dfeb778ed5a46ce624cf147f5887/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L429)):</sup>
I'd rather see a refactoring than code duplication, but that might be out of scope.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/175)

**Comment by harrison-tarr** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 429 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/175#-JzGTZnqTG60Z-qBJbFt)** ([raw file](https://github.com/codice/ddf/blob/c356b4c865a7dfeb778ed5a46ce624cf147f5887/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L429)):</sup>
i'll wrap it in a function when I fix the debugging. shouldn't take long at all :100: 

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/175)

**Comment by AzGoalie** [comment]
Full build passed :+1:

**Comment by shaundmorris** [comment]
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 417 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/175#-JzGvdMjfd14OWusCIpx)** ([raw file](https://github.com/codice/ddf/blob/c356b4c865a7dfeb778ed5a46ce624cf147f5887/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L417)):</sup>
You may also want to document this as well in the asciidoc as a note on queries through OpenSearch

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 430 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/175#-JzD5AgKwDsHpEkhhPKm-r1-430)** ([raw file](https://github.com/codice/ddf/blob/c356b4c865a7dfeb778ed5a46ce624cf147f5887/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L430)):</sup>
This might help a little bit with our documentation on logging: https://github.com/codice/ddf/blob/master/distribution/docs/src/main/resources/ExtContents.adoc#logging

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 444 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/175#-JzGTnJC2ZHOxyze8uFv)** ([raw file](https://github.com/codice/ddf/blob/c356b4c865a7dfeb778ed5a46ce624cf147f5887/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L444)):</sup>
:+1: 

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/175)

**Comment by harrison-tarr** [comment]
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 417 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/175#-JzGvdMjfd14OWusCIpx)** ([raw file](https://github.com/codice/ddf/blob/c356b4c865a7dfeb778ed5a46ce624cf147f5887/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L417)):</sup>
sure, where does that live?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/175)

**Comment by shaundmorris** [comment]
Review status: 0 of 2 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 417 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/175#-JzGvdMjfd14OWusCIpx)** ([raw file](https://github.com/codice/ddf/blob/c356b4c865a7dfeb778ed5a46ce624cf147f5887/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L417)):</sup>
https://github.com/codice/ddf/blob/master/catalog/docs/src/main/resources/IntContents.adoc#opensearch-endpoint

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/175)

**Comment by brendan-hofmann** [comment]
:+1: 

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/175)

File: distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java [hunk 3]
```diff
@@ -192,6 +192,18 @@ public void testFederatedQueryBySearchPhrase() throws Exception {
...
     /**
+     * Tests Source can retrieve based on a pure spatial query
+     * @throws Exception
+     */
+    @Test
+    public void testFederatedSpatial() throws Exception {
+        String queryUrl = OPENSEARCH_PATH + "?lat=10.0&lon=30.0&radius=250000&spatialType=POINT_RADIUS" + "&format=xml&src="
+                + OPENSEARCH_SOURCE_ID;
+        when().get(queryUrl).then().log().all().assertThat()
+                .body(containsString(RECORD_TITLE_1), containsString(RECORD_TITLE_2));
```
Comment (pklinef) [inline distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java]: I would suggest adding a negative case of a record that should not be returned for that query.  That will protect against a regression that just returns all results.

**Comment by harrison-tarr** [comment]
Review status: 1 of 2 files reviewed at latest revision, 5 unresolved discussions, some commit checks pending.

---

<sup>**[distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java, line 203 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/175#-JzD5AgL4hSTEwMpoM7b-r2-203)** ([raw file](https://github.com/codice/ddf/blob/307489863ccd8e16f634529a48a5629639f05aa5/distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java#L203)):</sup>
added and passes

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/175)

**Comment by AzGoalie** [comment]
Built and passed all tests
:+1:

**Comment by shaundmorris** [comment]
Got the same results as @AzGoalie

**Pull request merged**